torch/backends/_nnapi/serializer.py

```diff
@@ -957,23 +957,26 @@ def add_flatten(self, node):
             in_oper.shape[end_dim + 1:]
         )
 
-        # TODO(axit): To add support for runtime
-        # if any(dim == 0 for dim in in_oper.shape[start_dim: end_dim + 1]):
-        #     raise Exception("Flattened dims can't be flexible")
-        # non_flattened_dims = in_oper.shape[: start_dim] + in_oper.shape[end_dim + 1:]
-        # if non_flattened_dims.count(0) > 1:
-        #     raise Exception("Only 1 dim can be flexible")
-        # out_shape = tuple(
-        #     dim if dim != 0 else -1
-        #     for dim in out_shape
-        # )
+        if any(dim == 0 for dim in in_oper.shape[start_dim: end_dim + 1]):
+            raise Exception("Flattened dims can't be flexible")
```

Comment:
I would update this to say "Flattening flexible dims is not supported yet.", because it seems like it should be possible to support this with more work.